bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java

```diff
@@ -0,0 +1,368 @@
+
+/**
+ * Representation of a connection with a Bosch Smart Home Controller bridge.
+ *
+ * @author Stefan Kästle - Initial contribution
+ * @author Gerd Zanker - added HttpClient with pairing support
+ * @author Christian Oeing - refactorings of e.g. server registration
+ */
+@NonNullByDefault
+public class BoschSHCBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BoschSHCBridgeHandler.class);
+
+    private @Nullable BoschHttpClient httpClient;
+
+    private final Gson gson = new Gson();
+
+    /**
+     * Handler to do long polling.
+     */
+    private final LongPolling longPolling;
+
+    public BoschSHCBridgeHandler(Bridge bridge) {
+        super(bridge);
+
+        this.longPolling = new LongPolling(this.scheduler, this::handleLongPollResult, this::handleLongPollFailure);
+    }
+
+    @Override
+    public void initialize() {
+        // Read configuration
+        BoschSHCBridgeConfiguration config = getConfigAs(BoschSHCBridgeConfiguration.class);
+
+        if (config.ipAddress.isEmpty()) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No IP address set");
+            return;
+        }
+
+        if (config.password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No system password set");
+            return;
+        }
+
+        SslContextFactory factory;
+        try {
+            // prepare SSL key and certificates
+            factory = new BoschSslUtil(config.password).getSslContextFactory();
+        } catch (PairingFailedException e) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-ssl");
+            return;
+        }
+
+        // Instantiate HttpClient with the SslContextFactory
+        BoschHttpClient httpClient = this.httpClient = new BoschHttpClient(config.ipAddress, config.password, factory);
+
+        // Start http client
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create http connection to controller");
+            return;
+        }
+
+        // Initialize bridge in the background.
+        // Start initial access the first time
+        scheduleInitialAccess(httpClient);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+
+        // Stop long polling.
+        this.longPolling.stop();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * Schedule the initial access.
+     * Use a delay if pairing fails and next retry is scheduled.
+     */
+    private void scheduleInitialAccess(BoschHttpClient httpClient) {
+        scheduler.schedule(() -> initialAccess(httpClient), 15, TimeUnit.SECONDS);
```

Comment:
You need to keep of the Future returned here and make sure to cancel it when this handler is disposed.

Reply:
Storing the future and cancel it in dispose